resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/MountVolumesFeatureStep.scala

```diff
@@ -120,6 +122,20 @@ private[spark] class MountVolumesFeatureStep(conf: KubernetesConf)
   override def getAdditionalKubernetesResources(): Seq[HasMetadata] = {
     additionalResources.toSeq
   }
+
+  private def checkPVCClaimNameWhenMultiExecutors(claimName: String): Unit = {
+    val invalidClaimName =
+      if (!claimName.contains(PVC_ON_DEMAND) && !claimName.contains(ENV_EXECUTOR_ID)) true
+      else false
+
+    val executorInstances = conf.get(EXECUTOR_INSTANCES)
+    if (executorInstances.isEmpty) return
+    if (invalidClaimName && executorInstances.get > 1) {
+      throw new IllegalArgumentException("PVC ClaimName should contain " +
+        PVC_ON_DEMAND + " or " + ENV_EXECUTOR_ID +
+        " when multiple executors are required")
```

Comment:
The error message looks informative but please include **the current claim name** in the error message too.

Reply:
Have added the current claim name